contracts/SproutToken.sol

```diff
@@ -216,4 +216,19 @@ contract SproutToken is ERC20Burnable {
     function ecoBalanceOf(address _account) public view returns (uint256){
         return mStakeBalance[_account];
     }
+
+    // Returns the amount of tokens owned by the account without taking the current
+    // stake into account. This is required by the frontend so it doesn't count the
+    // generated amount twice.
+    function rawBalanceOf(address _account) public view returns (uint256) {
```

Comment:
Why don't we combine `rawBalanceOf` and `lastDepositAndMintTimestamps` into one function to avoid unnecessary web3 calls (usually quite slow & providers such as infura bill based on number of requests)

Reply:
Good point, will fix.